ext/odbc/php_odbc.c

```diff
@@ -2972,59 +3000,63 @@ PHP_FUNCTION(odbc_foreignkeys)
 
 	if (result->numcols > 0) {
 		if (!odbc_bindcols(result)) {
-			efree(result);
+			zval_ptr_dtor(return_value);
 			RETURN_FALSE;
 		}
 	} else {
 		result->values = NULL;
 	}
 	result->conn_ptr = conn;
 	result->fetched = 0;
-	RETURN_RES(zend_register_resource(result, le_result));
 }
 /* }}} */
 #endif /* HAVE_SOLID */
 
 /* {{{ Returns a result identifier containing information about data types supported by the data source */
 PHP_FUNCTION(odbc_gettypeinfo)
 {
+	zend_value_error("Nope");
+	RETURN_THROWS();
+
 	zval *pv_conn;
 	zend_long pv_data_type = SQL_ALL_TYPES;
 	odbc_result *result = NULL;
 	odbc_connection *conn;
 	RETCODE rc;
 	SQLSMALLINT data_type;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "r|l", &pv_conn, &pv_data_type) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|l", &pv_conn, odbc_connection_ce, &pv_data_type) == FAILURE) {
 		RETURN_THROWS();
 	}
 
+	printf("Hell noo");
```

Comment:
I'm gathering yes for the debug code :D

Reply:
Yeah :D I'm glad that I didn't use something more spicy 😅